### PR TITLE
Add Cask: Bambu Studio v01.03.01.01

### PR DESCRIPTION
**Pull request by duffpop**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/bambu-studio.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+cask "bambu-studio" do
+  version "01.03.01.01,20221126112354"
+  sha256 "5b7f92514f3b7b3770edd2491fd2106edaf9659c0319fd5017e02314aafa9f62"
+
+  url "https://github.com/bambulab/BambuStudio/releases/download/v#{version.csv.first.gsub!(/^0/, "").gsub!(/(\.0)/, ".")}/Bambu_Studio_mac-v#{version.csv.first}-#{version.csv.second}.dmg",
```
Comment (samford) [inline Casks/bambu-studio.rb:5]: Is there a specific reason for using the dmg file from GitHub? The homepage links to `https://public-cdn.bambulab.com/upgrade/studio/software/01.03.01.01/Bambu_Studio_mac-v01.03.01.01.dmg` as the download link.

Just from a livecheck perspective, it's dramatically easier to identify the version from the dmg link on the homepage compared to the complicated approach we're currently using to match version information from GitHub release asset HTML. [If we end up switching URLs, I can provide a suggestion for the `livecheck` block.]

Comment (duffpop) [reply Casks/bambu-studio.rb:5]: Howdy. I opted to use the GitHub Releases URL through habit & to keep in line with OSS goodness, but I'm more than happy to switch the URL to simplify the Cask.

Pop me your suggestion and I'll update the PR. Thanks you for your feedback!

File: Casks/bambu-studio.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+cask "bambu-studio" do
+  version "01.03.01.01,20221126112354"
```
Comment (samford) [inline Casks/bambu-studio.rb:2]:
```suggestion
  version "01.03.01.01"
```

File: Casks/bambu-studio.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  sha256 "5b7f92514f3b7b3770edd2491fd2106edaf9659c0319fd5017e02314aafa9f62"
+
+  url "https://github.com/bambulab/BambuStudio/releases/download/v#{version.csv.first.gsub!(/^0/, "").gsub!(/(\.0)/, ".")}/Bambu_Studio_mac-v#{version.csv.first}-#{version.csv.second}.dmg",
+      verified: "github.com/bambulab/BambuStudio/"
```
Comment (samford) [inline Casks/bambu-studio.rb:6]:
```suggestion
  url "https://public-cdn.bambulab.com/upgrade/studio/software/#{version}/Bambu_Studio_mac-v#{version}.dmg"
```

File: Casks/bambu-studio.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      # Identify the latest tag from the response's `location` header
+      latest_tag = File.basename(headers["location"])
+      next if latest_tag.blank?
+
+      # Fetch the assets list HTML for the latest tag and match within it
+      assets_page = Homebrew::Livecheck::Strategy.page_content(
+        @url.sub(%r{/releases/?.+}, "/releases/expanded_assets/#{latest_tag}"),
+      )
+      assets_page[:content]&.scan(regex)&.map { |match| "#{match[0]},#{match[1]}" }
+    end
```
Comment (samford) [inline Casks/bambu-studio.rb:26]:
```suggestion
    url :homepage
    regex(/href=.*?Bambu[._-]Studio[._-]mac[._-]v?(\d+(?:\.\d+)+)\.dmg/i)
```

**Review by samford — COMMENTED** [review]
It seems like the `sha256` is the same for the dmg file from the first-party CDN, so this should do the trick.

**Comment by miccal** [comment]
Thank you @duffpop and @samford.

**Review by miccal — APPROVED** [review]

**Pull request merged**